src/Scripting/CSharp/ScriptOptionsExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Scripting;
+
+namespace Microsoft.CodeAnalysis.CSharp.Scripting
+{
+    public static class ScriptOptionsExtensions
+    {
+        public static ScriptOptions WithLanguageVersion(this ScriptOptions options, LanguageVersion languageVersion)
+        {
+            if (options.ParseOptions is CSharpParseOptions csharpParseOptions && csharpParseOptions.LanguageVersion == languageVersion)
+                return options;
```

Comment:
I think it would be better to throw here if `options.ParseOptions` is non-null and not C# parse options. 
Consider we more language specific options. Then setting one of the options of one language would drop all of the options set by the other language.
Something like:

```C#
var parseOptions = 
  (options.ParseOptions is null) ? CSharpScriptCompiler.DefaultParseOptions :
  (options.ParseOptions is CSharpParseOptions existing) ? existing : 
  throw new InvalidOperationException(...));

return options.WithParseOptions(parseOptions.WithLanguageVersion(languageVersion)
```

Reply:
good idea, I added this. not sure what the localization process for errors is though, because the error message is now in the resources file and exists only in English